omim2obo/main.py

```diff
@@ -303,40 +304,61 @@ def omim2obo(use_cache: bool = False):
             phenotype_genes[p_mim].append({
                 'gene_id': gene_mim, 'phenotype_label': p_lab, 'mapping_key': p_map_key, 'mapping_label': p_map_lab})
 
+    self_ref_case = 0
     # - Add relations (subclass restrictions)
     for p_mim, assocs in phenotype_genes.items():
         for assoc in assocs:
             gene_mim, p_lab, p_map_key, p_map_lab = assoc['gene_id'], assoc['phenotype_label'], \
                 assoc['mapping_key'], assoc['mapping_label']
             evidence = f'Evidence: ({p_map_key}) {p_map_lab}'
 
-            # General skippable cases
+            # Skip: No phenotype or unknown defect
             # - not p_mim: Skip because not an association to another MIM (Provenance:
             #  https://github.com/monarch-initiative/omim/issues/78)
             # - p_map_key == '1': Skip because association w/ unknown defect (Provenance:
             #  https://github.com/monarch-initiative/omim/issues/79#issuecomment-1319408780)
             if not p_mim or p_map_key == '1':
                 continue
 
-            # Gene->Disease non-causal relationships
+            # Add restrictions: Gene->Disease non-causal relationships
             # - RO:0003302 docs: see MORBIDMAP_PHENOTYPE_MAPPING_KEY_PREDICATES
             if p_map_key != '3':  # 3 = 'causal'. Handled separately below.
                 g2d_pred = MORBIDMAP_PHENOTYPE_MAPPING_KEY_PREDICATES[p_map_key] if len(assocs) == 1 else RO['0003302']
                 add_subclassof_restriction_with_evidence(graph, g2d_pred, OMIM[p_mim], OMIM[gene_mim], evidence)
 
-            # Disease->Gene & Gene->Disease: Causal relationships
-            # - Skip non-causal cases
-            #  - 3: The molecular basis for the disorder is known; a mutation has been found in the gene.
+            # Skip non-causal cases
             if len(assocs) > 1 or p_map_key != '3' or not p2g_is_definitive(p_lab):
                 continue
-            #  - Digenic: Should technically be none marked 'digenic' if only 1 association, but there are.
+
+            # Log review cases
+            # - Digenic: Should technically be none marked 'digenic' if only 1 association, but there are.
             if 'digenic' in p_lab.lower():
                 # noinspection PyTypeChecker typecheck_fail_old_Python
                 REVIEW_CASES.append({
                     "classCode": 1,
-                    "classShortName": "causalD2gButMarkedDigenic",
-                    "value": f"OMIM:{p_mim}: {p_lab} (Gene: OMIM:{gene_mim})",
+                    "classShortName": "D2G: Disease-defining but marked digenic",
+                    "value": f"(Phenotype: {p_mim} {p_lab}) (Gene: {gene_mim})",
+                })
+            # -Self-referential cases
+            self_ref_assocs: List[Dict] = get_self_ref_assocs(p_mim, gene_phenotypes)
+            if self_ref_assocs:
+                self_ref_case += 1
+                REVIEW_CASES.append({
```

Comment:
Breaking my PR comment policy for a second.

Just adding that I reorganized/rephrased some comments here.

The ` REVIEW_CASES.append()` appears messy and WET (write everything twice) now. And I still have another review case to add, the one for handling unexpected MIM types. So when I do that PR, if you don't mind, I'll also refactor this code to be much cleaner; there will be an `add_review_case()` function.

Reply:
Gotcha, yes it would be good to refactor this section slightly so each case definition here is not both declared and appended here with it being more clean to have a structure that defines all the cases and then just append the case here as relevant.